### PR TITLE
chore: update go.mod and pipeline to use the same Go version of Dockerfile

### PR DESCRIPTION
**Pull request by thisthat**
## This PR

Align Go version to what is used in the Dockerfile

Integration test run: https://github.com/keptn/keptn/actions/runs/4283773729

### CODE REVIEW

File: webhook-service/lib/template_engine_test.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ func TestTemplateEngine_ParseTemplate(t1 *testing.T) {
 			},
 			want:    "",
 			wantErr: true,
-			errMsg:  "unexpected",
+			errMsg:  "template: :1: bad character U+007D '}'",
```
Comment (mowies) [inline webhook-service/lib/template_engine_test.go:48]: why is this change included?

Comment (thisthat) [reply webhook-service/lib/template_engine_test.go:48]: It looks like that with go 1.19 the std library to handle templating has better error messages

Comment (mowies) [reply webhook-service/lib/template_engine_test.go:48]: ok cool

**Review by mowies — APPROVED** [review]
